test/unit/app/managers/test_JobConnectionsManager.py

```diff
@@ -52,7 +56,7 @@ def setup_connected_dataset(sa_session: galaxy_scoped_session):
     return center_hda, expected_graph
 
 
-def setup_connected_dataset_collection(sa_session: galaxy_scoped_session):
```

Comment:
I've seen this a few places in the PR - I convinced myself it made sense in the TagHandler because it is shared with the tool shed - though we should axe the CommunityTagHandler and simplify that in another PR. I'm unsure about it here though. I'm curious if in general you think the scoped session is a better type than the galaxy_scoped_session? It seems like saying this is an SQLAlchemy session that connects to a Galaxy database is a more precise and useful type than the generic scoped session. Am I wrong though? Should the best practice be avoiding galaxy_scoped_session?

Reply:
@jmchilton These changes were needed because:
1) I added type annotation to `lib/galaxy/model/base.py:ModelMapping.__init__()`
2) mypy correctly inferred that `ModelMapping.session` is of type `scoped_session`
3) As a consequence, I had to correct the type annotation in `lib/galaxy/managers/context.py:ProvidesAppContext.sa_session` and then in all the other places.

Alternatively we could use a cast somewhere (e.g. in `lib/galaxy/model/mapping.py:GalaxyModelMapping`) to force the `galaxy_scoped_session`, but not sure it adds value?